tempesta_fw/pool.c

```diff
@@ -99,6 +126,26 @@ EXPORT_SYMBOL(tfw_pool_realloc);
 void
 tfw_pool_free(TfwPool *p)
 {
-	free_pages((unsigned long)p, p->order);
+	TfwPoolChunk *chunk, *next;
+
+	chunk = p->head;
+	while (chunk != NULL) {
+		next = chunk->next;
+		free_pages((unsigned long)chunk, chunk->order);
+		chunk = next;
+	}
 }
 EXPORT_SYMBOL(tfw_pool_free);
+
+void
+tfw_pool_try_free(TfwPool *p, void *ptr, size_t n)
+{
+	TfwPoolChunk *chunk;
+
+	chunk = __tfw_find_chunk(p, ptr);
+	if (ptr + n == chunk->base + chunk->off) {
+		chunk->off -= n;
+		memset(ptr, 0, n);
```

Comment:
Well, memset() and TFW_POOL_ZERO was a poor thing, and now it goes worse - ```tfw_pool_try_free()``` has sens only if it's extremely lightweight and fast, otherwise our pool goes to be common memory allocator. So we should not do the memory zeroing if we're not sure do we need this or not.

Please review _all_ memory allocation in the project. For example, tfw_http_msg_alloc() requests zeroed memory, by also calls memset() for just allocated memory.

It's good to create the pool w/o any flags, but extend tfw_pool_alloc() and tfw_pool_realloc() with the flag, so if the called needs zeroed memory, then it should request it only for the required memory chunk.